Sources_v3/StreamChatUI/Utils/ChatRouter.swift

```diff
@@ -0,0 +1,17 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import UIKit
+
+open class ChatRouter<Controller: UIViewController> {
```

Comment:
Decided to go with `ChatRouter` instead of `ChatCoordinator` because the latter usually has 2 responsibilities:
1. builds and shows the screen
2. handles the navigation from the screen

IMO better to follow the **SRP**

Another point is that `Coordinator` approach usually introduces parent-child relation between coordinators but currently it just complicates things. It's OK that the whole module graph is kept alive by the view-controller that is shown to the user (or lays somewhere in the navigation stack).

Reply:
Understood. We will have to think this through properly, ideally in a separate task. Can you create it, please?

Reply:
Here we are https://stream-io.atlassian.net/browse/CIS-429